Fix/issue 12033 url label

timwright12 opened the pull request:
## Description
Fixes https://github.com/WordPress/gutenberg/issues/12033

## How has this been tested?
Browser
Local Unit
Local e2e

## Screenshots <!-- if applicable -->
<img width="280" alt="screen shot 2018-12-17 at 3 52 18 pm" src="https://user-images.githubusercontent.com/459581/50114891-c8eb2a80-0213-11e9-8998-d172747b8b2b.png">


## Types of changes
Bug fix/suggestion to update the permalink label to be more descriptive 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

Commit:
merge from upstream

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
local save

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
darked focus state outline for block lists

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'fix/issue-12254-make-selected-block-outline-darker'

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
sasve

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
updated permalink labelling

jorgefilipecosta commented on package-lock.json:
```diff
@@ -11359,7 +11359,7 @@
 				"yargs": {
 					"version": "11.1.0",
-					"resolved": "http://registry.npmjs.org/yargs/-/yargs-11.1.0.tgz",
+					"resolved": "https://registry.npmjs.org/yargs/-/yargs-11.1.0.tgz",
```
I think this URL changes may be unintentional? If they are intentional maybe we can do the update to https in a separate PR.

timwright12 replied:
@jorgefilipecosta I think that just happened on the npm install, but I can manually revert it if it's a concern

jorgefilipecosta reviewed (commented):
Thank you for your contribution @timwright12. The label change seems to be working and now we see "URL Slug" as shown in your screen shot. I fear some users may not know what "slug" means but I don't have a better idea.
cc: @jasmussen, @karmatosed as they may be able to provide some ideas.

jasmussen commented:
I agree "slug" isn't the best word. "Link title (URL slug)" might work better?

timwright12 commented:
@jasmussen there are some good notes on naming in the issue from @afercia https://github.com/WordPress/gutenberg/issues/12033

This may be complicating the task, but what about a tooltip explaining what it is? To me, "link title" has the same problems as "url slug"

jasmussen commented:
Yep, I'm sure there's almost a decade of history here. But that's why I suggested the full title that included both. Open to suggestions, but since we're touching this it feels like an opportunity to be slightly more verbose.

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
reverted package lock

timwright12 commented:
@jasmussen I'm actually unfamiliar with, "Link title" as a term in that context, happy to update if others are, just want to confirm before.

@jorgefilipecosta I updated the package lock so there's no diff

jasmussen commented:
Here's a little quick research. Squarespace calls it URL slug, but has a big description below:

![screenshot 2018-12-21 at 08 05 23](https://user-images.githubusercontent.com/1204802/50329453-af89fd00-04f7-11e9-9ef0-6495c87d2dff.png)

I like the idea of a big description below regardless of what the label says.

Google Sites calls it "Custom Path":

![screenshot 2018-12-21 at 08 07 52](https://user-images.githubusercontent.com/1204802/50329468-bfa1dc80-04f7-11e9-836e-4bd6e382377e.png)

GoDaddy's site builder doesn't have this customization as an option.

Weebly calls it "Page Permalink" but also "Pages URL":

![screenshot 2018-12-21 at 08 13 17](https://user-images.githubusercontent.com/1204802/50329640-553d6c00-04f8-11e9-91a7-0d8cc5ee357f.png)

Wix calls it Page URL:

![screenshot 2018-12-21 at 08 18 42](https://user-images.githubusercontent.com/1204802/50329801-0fcd6e80-04f9-11e9-8946-74d425449e2c.png)

What about calling the label "Page URL" but then elaborating in a description below?

Commit:
updated label and added helper link

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

timwright12 commented:
@jorgefilipecosta I'm about to head out for the holidays but I didn't want to leave this sitting, so I updated the PR to use "Page URL" and added an external link to "Learn more about permalinks" under it. What do you think about that as a solution?

jasmussen commented:
Sorry about missing the added fix, and thank you for your work.

Screenshot of the current state:

<img width="279" alt="screenshot 2019-01-23 at 10 18 38" src="https://user-images.githubusercontent.com/1204802/51596039-5a774780-1ef8-11e9-82de-3455b409e84c.png">

Let's get a quick fix in and ship this as an improvement. But seeing this alone makes it clear that it doesn't work with just "Page URL" as the label, because we're just showing the last part. Here's what could probably work for now:

```
URL Slug:
[ test-page ]
The last part of the URL. Read more.
```

The help text below is italic, and read more links to https://codex.wordpress.org/Posts_Add_New_Screen.

Can we make that change and get this in?

afercia commented:
Can we avoid "read more" links please? Links should be meaningful even when read out of context.

jasmussen commented:
Sure. "Read about permalinks"?

timwright12 commented:
I can button this one up early next week, thanks for the feedback guys!

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg

Commit:
Merge branch 'master' into fix/issue-12033-URL-label

Commit:
post link text update

timwright12 commented:
@jasmussen @afercia This is all set!

Commit:
Merge branch 'master' of github.com:WordPress/gutenberg into fix/issue-12033-URL-label

timwright12 commented:
@jasmussen all set!

jasmussen commented:
Woot! Put it in a milestone and we'll get it in. All it needs is a final code sanity check now.

timwright12 commented:
@jasmussen I don't believe I have access to add anything to a milestone, but it looks like it's assigned to one already.

jasmussen commented:
Yep, I added it, so it's likely it'll land in the next plugin release.

timwright12 commented:
Perfect, thanks!

talldan commented on packages/edit-post/src/components/sidebar/post-link/index.js:
```diff
@@ -53,38 +53,46 @@ function PostLink( {
-							if ( ! forceEmptyField ) {
+				<div className="editor-post-link">
+					<TextControl
+						label={ __( 'URL Slug:' ) }
```
Hey @timwright12 - It looks like this accidentally changed back to 'URL Slug' in this commit https://github.com/WordPress/gutenberg/pull/12959/commits/6049de64108fc46e2d89c623bdefa9ad3c4b26ab

I think there was an agreement on 'Page URL', so I thought I'd flag it.

talldan replied:
Oh wait, no, I see this comment now:
https://github.com/WordPress/gutenberg/pull/12959#issuecomment-456729023

Will give it a proper test and review in that case 😄

talldan commented on packages/edit-post/src/components/sidebar/post-link/index.js:
```diff
@@ -53,38 +53,46 @@ function PostLink( {
-						// the field temporarily empty while typing.
-						if ( ! newValue ) {
-							if ( ! forceEmptyField ) {
+				<div className="editor-post-link">
```
It looks like the className `editor-post-link` doesn't have any associated styles. It can probably be removed unless there's a good reason for it?

youknowriad replied:
Is this div really necessary or should we just replace it with `<Fragment />`

talldan commented on packages/edit-post/src/components/sidebar/post-link/index.js:
```diff
@@ -53,38 +53,46 @@ function PostLink( {
-							if ( ! forceEmptyField ) {
+				<div className="editor-post-link">
+					<TextControl
+						label={ __( 'URL Slug:' ) }
```
Though my search wasn't exhaustive, it seems as though most input field labels in the sidebar don't end with a colon. The only other one I could see is 'Parent Page:'.

Might be good to drop the colon for sake of consistency. 

@jasmussen what do you think?

jasmussen replied:
That's an EXCELLENT catch & eye for detail. Yeah, if we can get that one final detail smoothed out let's ship it. Otherwise, this is also very lovely as is.

timwright12 replied:
@talldan @jasmussen removed the colon, nice catch!

talldan reviewed (commented):
Looks good! I had a couple of minor comments, which should be easy to resolve.

timwright12 committed:
Update index.js

jasmussen reviewed (approved):
Looks good!

<img width="300" alt="screenshot 2019-02-12 at 11 08 45" src="https://user-images.githubusercontent.com/1204802/52627842-975eaa80-2eb6-11e9-8b1c-684f82e2f60f.png">

Thanks for this.

I'd appreciate final eyes by @talldan or others, but let's ship this!
